### PR TITLE
Sort station observations at lower layer; Add latest observation

### PR DESCRIPTION
**Pull request by lymanepp**
Sort station observations to work around API bug https://github.com/weather-gov/api/discussions/474

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
This is probably the right place for this.  Should we revert the changes in #60?

File: pynws/nws.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ async def get_stations_observations(self, limit=0, start_time=None):
         res = await raw_stations_observations(
             self.station, self.session, self.userid, limit, start_time
         )
-        return [o["properties"] for o in res["features"]]
+        observations = [o["properties"] for o in res["features"]]
+        return sorted(observations, key=lambda o: o.get("timestamp"))
```
Comment (MatthewFlamm) [inline pynws/nws.py]: I think the default is to be in reverse sorted order?

Comment (lymanepp) [reply pynws/nws.py]: That might be. It's hard to tell right now when they're out of order. It's easy enough to reverse the order though.

Edit: the stations_observations_multiple.json test data has them sorted in ascending order.

Comment (lymanepp) [reply pynws/nws.py]: I was looking for the most recent observation--which is likely the most common use case.

I found that there's a specific URL for that: /stations/{stationId}/observations/latest

**Comment by lymanepp** [comment]
> This is probably the right place for this. Should we revert the changes in #60?

I noticed the test case, but I was a little confused that the Nws class was returning them out-of-order. I didn't think to look at the higher layer. It makes sense to me to have it in all layers (i.e., revert some of the changes in #60).

File: pynws/nws.py (hunk 1 above)
Comment (MatthewFlamm) [reply pynws/nws.py]: > stations_observations_multiple.json test data has them sorted in ascending order.

This data actually seems to be malformed for "timestamp".  The value is the same for both entries.  The other "multiple" is "unsorted" meaning it is out of order to the default NWS style.  I think the default is descending, i.e. newest first.

You can use `limit=1` to get only the last observation in the current code.  Do you think the other end point has utility beyond this usage?

Comment (lymanepp) [reply pynws/nws.py]: I tried using limit=1 but it didn't get the latest entry.

Comment (MatthewFlamm) [reply pynws/nws.py]: Ah this is a really bad bug on the NWS servers then.  I don't mind adding it to this library.

**Comment by lymanepp** [comment]
I added a method to retrieve the latest station observation to the API. Let me know what you think.

Note that I continued the general convention where the method name matches the NWS URI.

File: pynws/nws.py (hunk 1 above)
Comment (lymanepp) [reply pynws/nws.py]: > I think the default is to be in reverse sorted order?

Any suggestions on how to answer this definitively while their API is broken?

Comment (MatthewFlamm) [reply pynws/nws.py]: I can't answer definitely with a source, but an additional pointer is that this line selects the first valid observation.  This requires the observations to be sorted in descending order.  This worked with the default NWS response prior to #60.

https://github.com/MatthewFlamm/pynws/blob/7428f0a01e532c770d98889e6a64af1e8e7cf3a0/pynws/simple_nws.py#L271

**Comment by MatthewFlamm** [comment]
> I added a method to retrieve the latest station observation to the API. Let me know what you think.
> 
> Note that I continued the general convention where the method name matches the NWS URI.

LGTM. Thanks!

File: pynws/nws.py (hunk 1 above)
Comment (lymanepp) [reply pynws/nws.py]: That's good enough for me! I changed it to sort in descending order.

**Comment by MatthewFlamm** [comment]
This still LGTM if you are still OK with the ordering of the forecasts.  We need to clean up conflicts before merging.

**Comment by lymanepp** [comment]
> This still LGTM if you are still OK with the ordering of the forecasts. We need to clean up conflicts before merging.

The forecast ordering matches your original workaround in simple_nws.py. Now that it's handled in the lower layer, I removed the sorting from simple_nws.py. I fixed the merge conflicts.

File: pynws/raw_data.py [hunk 2]
```diff
@@ -43,6 +43,16 @@ async def raw_stations_observations(station, websession, userid, limit=0, start=
...
 
 
+async def raw_stations_observations_latest(station, websession, userid):
+    """Get observation response from station"""
+    url = pynws.urls.stations_observations_latest_url(station)
+    header = get_header(userid)
+    async with websession.get(url, headers=header) as res:
+        res.raise_for_status()
+        obs = await res.json()
+    return obs
```
Comment (MatthewFlamm) [inline pynws/raw_data.py]:
```suggestion
    return await _make_request(websession, url, header)
```

**Review by MatthewFlamm — COMMENTED** [review]
Thanks for removing the duplicated sort.  One last comment and it is good to go.

File: pynws/raw_data.py (hunk 2 above)
Comment (lymanepp) [reply pynws/raw_data.py]: Good catch! Fixed.

**Comment by MatthewFlamm** [comment]
Thanks!

**Pull request merged**